### PR TITLE
Add Python directory to 'PATH' environment variable

### PR DESCRIPTION
**Pull request by Siecje**
Fixes https://github.com/takluyver/pynsist/issues/90

### CODE REVIEW

**Comment by takluyver** [comment]
Thanks for this! I'll take your word that it works, and I can't think of any ill effects, though I'm still confused about why it's needed.

Specific thoughts:
1. I'd like to calculate the directory where Python lives using `sys.executable`, rather than from the install dir, so it's robust if Python.exe is placed somewhere else. We should check that this works correctly with the command-line exe entry points, though.
2. Let's have a comment briefly describing why we're adding to PATH.
3. In principle this is only needed for bundled Python (`format=bundled`), not when the Python installer is run separately. However, I'm thinking of dropping `format=installed` (#99), so this may not be a big concern.

**Comment by Siecje** [comment]
It shouldn't be needed, it should check the executable path for .dll files. Maybe this is a bug with the bare bones Windows 7 32-bit? I will test on Windows 8.1.

`sys.executable` does have the correct value, for me it is

`C:\Program Files\QMLApplication2\Python\pythonw.exe`

**Comment by takluyver** [comment]
I'm confident `sys.executable` will be right from the GUI shortcuts; have you also checked it with the new [command definitions](http://pynsist.readthedocs.io/en/latest/cfgfile.html#command-sections)? I'm pretty sure it will work there too, but I'd like a check.

**Comment by Siecje** [comment]
I have not checked `command definitions` or `format=installed`.

File: nsist/commands.py [hunk 1]
```diff
@@ -11,6 +11,9 @@
 sys.path.insert(0, pkgdir)
 os.environ['PYTHONPATH'] = pkgdir + os.pathsep + os.environ.get('PYTHONPATH', '')
 
+# Allowing .dll files in Python directory to be found
+os.environ['PATH'] += ';' + os.path.dirname(sys.executable)
```
Comment (takluyver) [inline nsist/commands.py]: I started wondering if it's possible for PATH not to be set at all. It is, but you can't even start Python itself in that case, so I don't think we need to worry about it.

**Comment by Siecje** [comment]
I just tried with a `command definitions` and it works. I'm not sure how though since the `sys.executable` is in the `bin/` directory.

**Comment by takluyver** [comment]
I think the exe in `bin/` actually runs `python.exe` as a subprocess, so `sys.executable` will be correct. I'll just double check that...

**Comment by takluyver** [comment]
Yep, I checked that using my Windows VM, and that is how it works.

**Pull request merged**